specification/cognitiveservices/data-plane/Language/preview/2022-03-01-preview/analyzeconversations.json

```diff
@@ -0,0 +1,1430 @@
+          }
+        },
+        "entities": {
+          "description": "The entity extraction results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationEntity"
+          }
+        }
+      }
+    },
+    "ConversationIntent": {
+      "type": "object",
+      "description": "The intent classification result of a Conversation project.",
+      "required": [
+        "category",
+        "confidenceScore"
+      ],
+      "properties": {
+        "category": {
+          "description": "A predicted class.",
+          "type": "string"
+        },
+        "confidenceScore": {
+          "format": "float",
+          "x-ms-client-name": "confidence",
+          "description": "The confidence score of the class from 0.0 to 1.0.",
+          "type": "number",
+          "minimum": 0,
+          "maximum": 1
+        }
+      }
+    },
+    "ConversationEntity": {
+      "type": "object",
+      "description": "The entity extraction result of a Conversation project.",
+      "required": [
+        "category",
+        "text",
+        "offset",
+        "length",
+        "confidenceScore"
+      ],
+      "properties": {
+        "category": {
+          "description": "The entity category.",
+          "type": "string"
+        },
+        "text": {
+          "description": "The predicted entity text.",
+          "type": "string"
+        },
+        "offset": {
+          "format": "int32",
+          "description": "The starting index of this entity in the query.",
+          "type": "integer"
+        },
+        "length": {
+          "format": "int32",
+          "description": "The length of the text.",
+          "type": "integer"
+        },
+        "confidenceScore": {
+          "format": "float",
+          "x-ms-client-name": "confidence",
+          "description": "The entity confidence score.",
+          "type": "number"
+        },
+        "resolutions": {
+          "description": "The collection of entity resolution objects.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BaseResolution"
+          }
+        },
+        "extraInformation": {
+          "description": "The collection of entity extra information objects.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BaseExtraInformation"
+          }
+        }
+      }
+    },
+    "BaseExtraInformation": {
+      "description": "The abstract base object for entity extra information.",
+      "type": "object",
+      "discriminator": "extraInformationKind",
+      "properties": {
+        "extraInformationKind": {
+          "description": "The extra information object kind.",
+          "type": "string",
+          "enum": [
+            "EntitySubtype",
+            "ListKeys"
+          ],
+          "x-ms-enum": {
+            "name": "ExtraInformationKind",
+            "modelAsString": false
```

Comment:
This will *never* be used for output? If any doubt, make `true`.

Reply:
Actually, I see that `BaseExtractInformation` is returned by `ConversationEnttity`, which is described as, "The entity extraction result of a Conversation project." Since it's a result (a response), `modelAsString` must be `true`. This is true for any model or model property type that can be in a result. This will break SDKs if you ever add a value in a future version.

`modelAsString: false` (or the absence of the property) is only valid for purely input models - and ones that would only ever be input models. If you ever needed to make such a model also output, you'd need to define a new model. Changing `modelAsString` is a breaking change for the SDKs (for most languages, anyway).

Reply:
Or, if you're 100% sure some of these will never change (like units; I'd be skeptical if result discriminators weren't expected to change in time), you can leave this `false`. See https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums for details.

Reply:
I think it should be true. Let me confirm it with @fxs130430 .

Reply:
Confirmed and it should be true.